plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/AbstractCommandChange.java

```diff
@@ -66,6 +67,18 @@ public abstract class AbstractCommandChange<T extends EObject> extends Change {
 	 */
 	protected AbstractCommandChange(final URI elementURI, final Class<T> elementClass) {
 		this(elementURI.lastSegment(), elementURI, elementClass);
+		initEnablement();
+	}
+
+	/**
+	 * It needs to be checked whether a change has been set to "NO_CHANGE"
+	 *
+	 */
+	private void initEnablement() {
+		if ((this instanceof final IFordiacPreviewChange change)
+				&& change.getDefaultSelection().contains(ChangeState.NO_CHANGE)) {
+			setEnabled(false);
+		}
```

Comment:
I believe we should avoid introducing behavior for specific subclasses in the base class. I suggest either doing this only in the relevant subclasses or maybe introduce a common abstract class for preview changes?

Reply:
you are right, i will make the ChangePreviewer abstract and make it as super class of the AbstractCommandChange.
If a change does not need a preview that it should also be fine, since they first have to be defined in the plugin.xml

Reply:
i meant the other way . the new abstract class needs to inherit AbstractCommandChange.